modules/javafx.controls/src/main/java/javafx/scene/control/skin/SliderSkin.java

```diff
@@ -389,6 +393,24 @@ public Object queryAccessibleAttribute(AccessibleAttribute attribute, Object...
                     cur.getX() - dragStart.getX() : -(cur.getY() - dragStart.getY());
             behavior.thumbDragged(me, preDragThumbPos + dragPos / trackLength);
         });
+
+        thumb.setOnMouseEntered(me -> {
+            Tooltip t = getSkinnable().getTooltip();
+            if (t != null && t.isAutoHide()) {
+                tooltipConsumeAutoHidingEvents = t.getConsumeAutoHidingEvents();
+                // Temporarily disable consuming auto hiding events from tooltip.
+                // This is done to receive mouse pressed event on thumb and
+                // inturn to detect start of drag.
+                t.setConsumeAutoHidingEvents(false);
```

Comment:
In general, I'm not in favor of setting the value of a public property (for a few reasons). Have you considered alternative approaches? One thing to note is that this will be ineffective if the app binds to the `consumeAutoHidingEvents` property (unlikely as that might be).

Reply:
I considered following 2 approaches.

1.  Initializing `dragStart` in `layoutChildren` method as suggested in the bug. With this approach I was not able to get the exact mouse click position on the thumb. Thumb was lagging behind the cursor while dragging and thumb was also flickering when initialized with (`layoutX`, `layoutY`) values.
2. Initializing `dragStart` in `initialize` method. In this case also I was not able to get the mouse click position on the thumb. Tried initializing thumb with `layoutX` and `layoutY` values, this causes side effect of cursor moving to the (`layoutX`, `layoutY`) on dragging the thumb.
Because of these side effects, considered above approach.
Please let me know if you have any suggestions.

Reply:
Thanks for the explanation. I don't have a better suggestion than what you propose, so let's proceed. It might be good to check whether the property is bound and skip setting it in the unlikely event that it is (save/restore the binding might be tricky and not something I can recall ever doing).

Reply:
Perhaps we can introduce another property - either public or internal to suppress consumption of auto-hiding events?

The other aspect of this problem, as we discussed earlier, is that, in my opinion, the tooltips should really be transparent to mouse clicks.  We do have this problem in Swing - if the tool tip overlaps a button , the user needs two clicks - first dismisses the tooltip and the other depresses the button.  That will be a behavioral change.

What do you think?

Reply:
Whatever we decide for this particular bug fix, I agree that we should reexamine whether tooltips should be transparent to mouse clicks by default.

As for this bug fix, I wouldn't want to introduce a new public property, so that leaves either the currently proposed solution of saving / restoring the existing public property, or adding an internal property. If this new internal property could be somehow limited to Tooltip, that might be OK; otherwise it could be pretty intrusive.

Reply:
Updated code to check whether `consumeAutoHidingEventsProperty` is bound and skip the property value change in mouse entered and exited event handlers.
Since we need to recheck the behavior of tooltip, I would like to keep the above proposed solution for the time being.